src/test/java/edu/ucla/library/services/utility/PropsWriterTest.java

```diff
@@ -0,0 +1,126 @@
+
+package edu.ucla.library.services.utility;
+
+import static com.github.stefanbirkner.systemlambda.SystemLambda.catchSystemExit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.junit.contrib.java.lang.system.SystemErrRule;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+
+/*
+ * Unit tests for ProsWriter.java
+ */
+public class PropsWriterTest {
+
+  /**
+   * default properties file.
+  */
+    private static final String DEFAULTS =
+        "src/test/resources/properties.default";
+
+  /**
+   * output template file.
+  */
+    private static final String TEMPLATE =
+        "src/test/resources/properties.tmpl";
+
+  /**
+   * final output file.
+  */
+    private static final String OUTPUT =
+        "src/test/resources/output.properties";
+
+  /**
+   * fake file name used to cause app failures.
+  */
+    private static final String FAKE =
+        "src/test/resources/not.there";
+
+  /**
+   * rule that allows catching System.err output in tests.
+  */
+    @Rule
+    public final SystemErrRule mySystemErrRule = new SystemErrRule().enableLog();
+
+  /**
+   * Sets up a PropsWriter test.
+   *
+   * @throws IOException If there is a problem deleting the file
+  */
+    @Before
+    public void setUp() throws IOException {
+        // delete the output file if lest over from previous run
+        Files.deleteIfExists(FileSystems.getDefault().getPath(OUTPUT));
+    }
+
+  /**
+   * Tests happy path for app execution.
+   *
+  */
+    @Test
+    public void testBuildProperties() {
+        PropsWriter.buildProperties(DEFAULTS, TEMPLATE, OUTPUT);
+        assertTrue(Files.exists(FileSystems.getDefault().getPath(OUTPUT)));
+    }
+
+  /**
+   * Tests that app catches nonexistent files.
+   *
+  */
+    @Test
+    public void testCheckExists() throws Exception {
+        final int statusCode = catchSystemExit(() -> {
+            PropsWriter.checkExists(FAKE);
+        });
+        assertTrue(mySystemErrRule.getLog().contains("File must exist"));
+        assertEquals(Constants.DOESNT_EXIST_ERROR, statusCode);
+    }
+
+  /**
+   * Tests that app catches nonreadable files.
+   *
+  */
+    @Test
+    public void testCheckReadable() throws Exception {
```

Comment:
I'm not sure how useful this test is -- seems to me to conflate the "readable-ness" of a file with its existence. I wonder if it makes sense to create a new file for test purposes that is not readable (for whatever reason), but exists.

Reply:
Probably missing something, but adding an unreadable file (no read permissions for owner/group/other) to test resources causes Maven error: 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.1.0:testResources (default-testResources) on project services-utilities: /home/cloud_user/Work/Utilities/src/test/resources/properties.locked (Permission denied)
```

Reply:
Maybe you could set the permissions programmatically from within the test method?

Reply:
It looks like the Maven error comes when it tries to move the test file into `target`. You can point the test to the file as it exists in the `src/test/resources` directory and then exclude the processing of that file in the testResources config in the POM so that Maven doesn't try to move it.

Actually, does the testResource config in the POM serve a purpose? You can just access those files where they are from the tests. Usually you'd want to move the test resources when there is a logback.xml config, or something, that needs to be in the test classpath. I'm not sure if that's necessary here(?)

Reply:
Went the programmatic route; making file unreadable via file system and excluding in `pom` works for running tests... but then `git` can't add the file to the repo.